tests/yarn_files/formatting.json

```diff
@@ -0,0 +1,132 @@
+[
+	{
+		"title": "Start",
+		"tags": "",
+		"body": "<<set $traveler to \"Adventurer\">>\n<<set $class to \"Rogue\">>\n<<set $gold to 20>>\n\nHello {traveler}! who's your day going ?",
+		"position": {
+			"x": 38,
+			"y": -120
+		},
+		"colorID": 0
+	},
+	{
+		"title": "Tavern",
+		"tags": "",
+		"body": "You have {gold, plural, =0 {no gold, what a shame...} one {one piece inside your pocket.} other {# gold pieces!} }",
+		"position": {
+			"x": 442,
+			"y": -124
+		},
+		"colorID": 1
+	},
+	{
+		"title": "LowMoney",
+		"tags": "",
+		"body": "<<set $gold to 1>>\n[[Tavern]]",
+		"position": {
+			"x": 442,
+			"y": -415
+		},
+		"colorID": 1
+	},
+	{
+		"title": "NoMoney",
+		"tags": "",
+		"body": "<<set $gold to 0>>\n[[Tavern]]",
+		"position": {
+			"x": 201,
+			"y": -415
+		},
+		"colorID": 1
+	},
+	{
+		"title": "HighMoney",
+		"tags": "",
+		"body": "<<set $gold to 175>>\n[[Tavern]]",
+		"position": {
+			"x": 693,
+			"y": -415
+		},
+		"colorID": 1
+	},
+	{
+		"title": "Course",
+		"tags": "",
+		"body": "You are {position, selectordinal, one {#st} two {#nd} few {#rd} other {#th}} to reach the final line!",
+		"position": {
+			"x": -440,
+			"y": -121
+		},
+		"colorID": 2
+	},
+	{
+		"title": "FirstCourse",
+		"tags": "",
+		"body": "<<set $position to 1>>\n[[Course]]",
+		"position": {
+			"x": -702,
+			"y": -511
+		},
+		"colorID": 2
+	},
+	{
+		"title": "ThirdCourse",
+		"tags": "",
+		"body": "<<set $position to 3>>\n[[Course]]",
+		"position": {
+			"x": -440,
+			"y": -511
+		},
+		"colorID": 2
+	},
+	{
+		"title": "NoHope",
+		"tags": "",
+		"body": "<<set $position to 999>>\n[[Course]]",
+		"position": {
+			"x": -202,
+			"y": -511
+		},
+		"colorID": 2
+	},
+	{
+		"title": "GenderMale",
+		"tags": "",
+		"body": "<<set $gender to \"male\">>\n[[Gender]]",
+		"position": {
+			"x": -239,
+			"y": 215
+		},
+		"colorID": 8
+	},
+	{
+		"title": "Gender",
+		"tags": "",
+		"body": "Note: {gender, select, male {He} female {She} other {They}}'ll remember that.",
```

Comment:
this formatting is slightly different than 
https://yarnspinner.dev/docs/syntax#expression 

`I haven't seen [select {$gender} m="him" f="her" nb="them"].`
but at least it gives us feature parity. I wonder if we can adjust MessageFormat's text parsing later on.

I am happy with just having the feature available to js users for now,so dont worry about it in this PR if it requires a lot of work

Reply:
because, it's using the naming convention for var, so un C# it's `$var`, but un js it's just `var`

Reply:
i saw that we can make custom template on the librairy, i'll try looking at the documentation

Reply:
[here](https://messageformat.github.io/messageformat/custom-formatters/)

Reply:
I wonder if yarnSpinner devs set it up in a special way in their source code, or this just comes from the formater being a javascript implementation

Reply:
I saw that this lib is on a lot of language, maybe devs wanted to implement it with the appropriate syntax following the language

Reply:
This is the commit which added format functions to yarnSpinner
https://github.com/YarnSpinnerTool/YarnSpinner/commit/c6e375675a066b85e7a447dd093d1a10df39fd6b

maybe it can help to figure out what they did

Reply:
Posted a question about this, maybe the devs can help us here
https://github.com/messageformat/messageformat/issues/318